### ISSUE TITLE
Inconsistent warnings and defaults for ixsep etc

### ISSUE THREAD

**Issue by JosephThomasParker** [boutproject/BOUT-dev#1753]
When the input file doesn't specify `ixsep1`, the output gives the contradictory warnings:
```
Variable 'ixseps1' not in mesh options. Setting to 0
        WARNING: Separatrix location 'ixseps1' not found. Setting to 12
```
(where `NxGlobal = 12`), and similarly for `ixseps2`, the `jyseps` and `ny_inner`.  This is because of how the checking is done in `include/bout/boutmesh.cxx`:
```
  // separatrix location
  if (Mesh::get(ixseps1, "ixseps1")) {
    ixseps1 = GlobalNx;
    output_warn.write(
        "\tWARNING: Separatrix location 'ixseps1' not found. Setting to %d\n", ixseps1);
  }
```
Here `Mesh::get` issues the first warning and sets the default to 0, then boutmesh overrides this.

I'm not sure if `Mesh::get` is used elsewhere, but perhaps the solution is to add an optional  "suppress_output" argument (default: false) to `Mesh::get` so that it doesn't write output when we're really only using it to see if the variable is set.

**Comment by bendudson** [comment]
Would another possibility be to add an `isSet` method, like `Options` has, to test if a variable is present before trying to get its value? 
Possibility number three is to pass a default value to the call to `Mesh::get``

**Comment by ZedThree** [comment]
Is there a possibility to replace `Mesh::get` and/or `GridSource` entirely with `Options` or `OptionsNetCDF`? I think `GridFile` can handle reading 3D variables stored as FFTs which `OptionsNetCDF` can't do (yet), so it's definitely not currently a drop-in replacement. Might be worth investigating though?

**Comment by bendudson** [comment]
Yes, I think the plan is to replace `DataFIle` with `Options`, reading using `OptionsNetCDF` etc. then a grid would be created by just passing in an Options node. I'd like to add support for HDF5 or other format, and then replace `DataFile`.

**Comment by JosephThomasParker** [comment]
> Would another possibility be to add an `isSet` method, like `Options` has, to test if a variable is present before trying to get its value?
> Possibility number three is to pass a default value to the call to `Mesh::get`

Number two works, but I think number three ends up being neater.  It looks like `Mesh::get` already accepted default values for Fields.

**Comment by ZedThree** [comment]
Fixed in #1754, although indentation of the message is a bit higgledy-piggledy
